massa-grpc/src/stream/new_endorsements.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 MASSA LABS <info@massa.net>
+
+use crate::error::{match_for_io_error, GrpcError};
+use crate::server::MassaGrpc;
+use futures_util::StreamExt;
+use massa_proto::massa::api::v1 as grpc;
+use std::io::ErrorKind;
+use std::pin::Pin;
+use tokio::select;
+use tonic::codegen::futures_core;
+use tonic::{Request, Streaming};
+use tracing::log::{error, warn};
+
+/// Type declaration for NewEndorsements
+pub type NewEndorsementsStream = Pin<
+    Box<
+        dyn futures_core::Stream<Item = Result<grpc::NewEndorsementsResponse, tonic::Status>>
+            + Send
+            + 'static,
+    >,
+>;
+
+/// Creates a new stream of new produced and received endorsements
+pub(crate) async fn new_endorsements(
+    grpc: &MassaGrpc,
+    request: Request<Streaming<grpc::NewEndorsementsRequest>>,
+) -> Result<NewEndorsementsStream, GrpcError> {
```

Comment:
Using a generic type can improve performance by allowing the compiler to generate more efficient code for each concrete type used with the function, rather than relying on dynamic dispatch to call the trait method. However, using a generic type can also make the code more complex and harder to read, especially if there are many different concrete types used with the function.

In our case, we have only one concret type which makes generic overkill.